### PR TITLE
feat: add fast-benchmarks

### PR DESCRIPTION
**Pull request by prudepixie**
# Pull Request

cherry pick commits as there were too many conflicts to resolve between features/fast-element-2
includes all changes from https://github.com/microsoft/fast/pull/5888

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: .github/workflows/ci-benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+name: Run Benchmarks
+
+on: [pull_request]
```
Comment (awentzel) [inline .github/workflows/ci-benchmarks.yml:3]: align with existing usage across pipelines

File: .github/workflows/ci-benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              yarn install
+              yarn run benchmark --library=fast-element --benchmark=render --versions latest local --localBenchFile=index2 --branchName=${GITHUB_HEAD_REF} --operations=update10th
+        # Upload this benchmarks results
+        - name: Upload results
+          uses: actions/upload-artifact@v2
+          with:
+              name: results
+              path: packages/utilities/fast-benchmarks/results/fast-element_render-update10th.json
+    render_createDelete5x:
+      name: Run render create and delete 5 times benchmark
```
Comment (awentzel) [inline .github/workflows/ci-benchmarks.yml:55]: Render create and delete 5X

File: .github/workflows/ci-benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+              yarn install
+              yarn run benchmark --library=fast-element --benchmark=render --versions latest local --localBenchFile=index2 --branchName=${GITHUB_HEAD_REF} --operations=create10k
+        # Upload this benchmarks results
+        - name: Upload results
+          uses: actions/upload-artifact@v2
+          with:
+              name: results
+              path: packages/utilities/fast-benchmarks/results/fast-element_render-create10k.json
+    render_update10th:
+      name: Run render update 10th benchmark
```
[the remark this answers is not in the document]
Comment (prudepixie) [reply .github/workflows/ci-benchmarks.yml]: 'Render' is the benchmark name, I've updated the description to make it more clear

**Comment by prudepixie** [comment]
All work is migrated to https://github.com/microsoft/fast/pull/6066

**Pull request closed without merging**